src/language.rs

```diff
@@ -80,3 +94,75 @@ impl Display for Language {
         }
     }
 }
+
+/// Users preferences for locales
+///
+/// Resulting fields are sorted in order of the user's preference.
+///
+/// POSIX locale values and GNU nonstandard categories defined in
+/// <https://man7.org/linux/man-pages/man5/locale.5.html>.
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct LanguagePrefs {
+    /// Determines general user language preference, should be used in
+    /// situations which are not encompassed by other [`LanguagePrefs`].
+    pub lang: Vec<Language>,
+
+    /// Determines collation rules used for sorting and regular expressions,
+    /// including character equivalence classes and multicharacter collating
+    /// elements.
+    pub collate: Vec<Language>,
+
+    /// Determines the interpretation of byte sequences as characters (e.g.,
+    /// single versus multibyte characters), character classifications (e.g.,
+    /// alphabetic or digit), and the behavior of character classes.
+    pub ctype: Vec<Language>,
+
+    /// determines the formatting used for monetary-related numeric values,
+    /// i.e, the way numbers are usually printed with details such as
+    /// decimal point versus decimal comma
+    pub monetary: Vec<Language>,
+
+    /// Determines the format and values of affirmative and negative responses
+    pub messages: Vec<Language>,
+
+    /// Determines the formatting rules used for nonmonetary numeric values.
+    /// For example, the thousands separator and the radix character
+    pub numeric: Vec<Language>,
+
+    /// Determines format and contents of date and time information
+    pub time: Vec<Language>,
+}
+
+impl Display for LanguagePrefs {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let format_langs = |langs: &[Language]| {
+            langs
+                .iter()
+                .map(ToString::to_string)
+                .collect::<Vec<_>>()
+                .join(":")
+        };
+        writeln!(f, "LANG=\"{}\"", format_langs(&self.lang))?;
+        writeln!(f, "LC_COLLATE=\"{}\"", format_langs(&self.collate))?;
+        writeln!(f, "LC_CTYPE=\"{}\"", format_langs(&self.ctype))?;
+        writeln!(f, "LC_MESSAGES=\"{}\"", format_langs(&self.messages))?;
+        writeln!(f, "LC_MONETARY=\"{}\"", format_langs(&self.monetary))?;
+        writeln!(f, "LC_NUMERIC=\"{}\"", format_langs(&self.numeric))?;
+        write!(f, "LC_TIME=\"{}\"", format_langs(&self.time))
```

Comment:
I think the formatting here is probably not what we want (too specific to environment variable names).

This implementation doesn't need to be in this PR, but the examples should use the `Display` implementation if it exists.

Perhaps something like:

```
Collation=en/US,de/DE;CharClass=en/US,Messages=en/US,Monetary=en/US;Numeric=en/US,Time=en/US
```